src/application.js

```diff
@@ -75,10 +76,11 @@ export default {
       });
 
     const hasMethod = methods => methods.some(name =>
-      (service && typeof service[name] === 'function'));
+      (service && typeof service[name] === 'function')
+    );
 
     // Check for service (any object with at least one service method)
-    if(hasMethod(['handle', 'set']) || !hasMethod(this.methods)) {
+    if(hasMethod(['handle', 'set']) || !hasMethod(this.methods.concat('setup'))) {
```

Comment:
Do we want to throw an error or a debug warning in case the service doesn't have a `setup` method?